tests/suites/test_suite_ssl.data

```diff
@@ -64,6 +64,9 @@ ssl_mock_tcp_interleaving:0:0x0FB1:0
 Test mock non-blocking TCP connection: both peers would block (interleaving)
 ssl_mock_tcp_interleaving:0:0x1111:0xEEEE
 
+Test handshake with different states
```

Comment:
This test name is a bit misleading. You don't really want to test the handshake; You'd like to test whether your state switching works. I would suggest `Test handshake state switching between x and y` for a pair of states, if you'll be implementing Jaeden's suggestion. In any case, a more descriptive name would be appreciated :)